torchtune/models/convert_weights.py

```diff
@@ -260,15 +260,21 @@ def tune_to_peft_adapter_weights(
     # re-use the _FROM_HF mapping for base model weights. We iterate over it twice:
     # once to add mappings for LoRA A matrices and once to add mappings for LoRA B matrices.
     for k, v in _TO_PEFT_KEYS.items():
-        full_mapping.update(
-            {
-                vv.replace(".weight", f".{k}.weight"): kk.replace(
-                    ".weight", f".{v}.weight"
+        for kk, vv in _FROM_HF.items():
+            if vv is None:
+                continue
+            if "lora" in k:
+                full_mapping.update(
+                    {
+                        vv.replace(".weight", f".{k}.weight"): kk.replace(
+                            ".weight", f".{v}.weight"
+                        )
+                    }
+                )
+            else:
+                full_mapping.update(
+                    {vv.replace(".weight", f".{k}"): kk.replace(".weight", f".{v}")}
                 )
-                for kk, vv in _FROM_HF.items()
-                if vv is not None
-            }
```

Comment:
Is this just for catching the DoRA magnitude parameter which should map to `{layer}.magnitude` rather than `{layer}.magnitude.weight`? Or are there any other cases to catch? 
If so why not change the control flow to explictly catch this case so it's super clear why we need different logic. If not, ignore me : )

Reply:
Yes that's correct. You mean changing it to something like `if k == "magnitude": (second branch of current code) else: (first branch of current code)`? Or do you want me to also modify the specific form of the dict item (`{vv.replace(".weight", f".{k}"): kk.replace(".weight", f".{v}")}`)?

Reply:
> You mean changing it to something like if k == "magnitude": (second branch of current code) else: (first branch of current code)

I think this sounds good! If you wanted to be clever about it you could do something like
```python